server_response.go

```diff
@@ -172,8 +171,22 @@ func SendProblemResponse(w http.ResponseWriter, r *http.Request, statusCode int,
 	return
 }
 
+func acceptContentType(r *http.Request, contentType string) bool {
+	if !strings.HasPrefix(contentType, "application/") {
```

Comment:
What happens, if client sent `Accept: text/plain` and server calls `NewErrorWithBody(nil, http.StatusBadRequest, "text/plain", "invalid field")`?

Reply:
I'd check exact match with base content type here. Or go on if `Accept` header is missing completely.

Reply:
sendCustomResponse and acceptContentType functions are updated